vcbuild.bat

```diff
@@ -376,55 +376,63 @@ goto cpplint
 if not defined cpplint goto jslint
 echo running cpplint
 set cppfilelist=
-setlocal enabledelayedexpansion
-for /f "tokens=*" %%G in ('dir /b /s /a src\*.c src\*.cc src\*.h ^
-test\addons\*.cc test\addons\*.h test\cctest\*.cc test\cctest\*.h ^
-test\gc\binding.cc tools\icu\*.cc tools\icu\*.h') do (
-  set relpath=%%G
-  set relpath=!relpath:*%~dp0=!
-  call :add-to-list !relpath!
-)
-( endlocal
-  set cppfilelist=%localcppfilelist%
-)
+set cpp_locs=src\*.c
+set cpp_locs=%cpp_locs% src\*.cc
+set cpp_locs=%cpp_locs% src\*.h
+set cpp_locs=%cpp_locs% test\addons\*.cc
+set cpp_locs=%cpp_locs% test\addons\*.h
+set cpp_locs=%cpp_locs% test\cctest\*.cc
+set cpp_locs=%cpp_locs% test\cctest\*.h
+set cpp_locs=%cpp_locs% test\gc\binding.cc
+set cpp_locs=%cpp_locs% tools\icu\*.cc
+set cpp_locs=%cpp_locs% tools\icu\*.h
+for /f "tokens=*" %%G in ('dir /b /s /a:, %cpp_locs%') do call :add-to-list %%G
 python tools/cpplint.py %cppfilelist%
 python tools/check-imports.py
 goto jslint
 
 :add-to-list
-echo %1 | findstr /c:"src\node_root_certs.h"
-if %errorlevel% equ 0 goto exit
-
-echo %1 | findstr /c:"src\queue.h"
-if %errorlevel% equ 0 goto exit
-
-echo %1 | findstr /c:"src\tree.h"
-if %errorlevel% equ 0 goto exit
-
-@rem skip subfolders under /src
-echo %1 | findstr /r /c:"src\\.*\\.*"
-if %errorlevel% equ 0 goto exit
-
-echo %1 | findstr /r /c:"test\\addons\\[0-9].*_.*\.h"
-if %errorlevel% equ 0 goto exit
-
-echo %1 | findstr /r /c:"test\\addons\\[0-9].*_.*\.cc"
-if %errorlevel% equ 0 goto exit
-
-set "localcppfilelist=%localcppfilelist% %1"
+setlocal enabledelayedexpansion
+set base_path=
+set relpath=%1
+set relpath=!relpath:%~dp0=!
+endlocal&set relpath=%relpath%
+@rem findstr can do multiple searches, but only for regexes
+set find_arg=/r
+set find_arg=%find_arg% /c:"src\\node_root_certs\.h"
+set find_arg=%find_arg% /c:"src\\queue\.h"
+set find_arg=%find_arg% /c:"src\\tree\.h"
+set find_arg=%find_arg% /c:"src\\.*\\.*"
+set find_arg=%find_arg% /c:"test\\addons\\[0-9].*_.*\.h"
+set find_arg=%find_arg% /c:"test\\addons\\[0-9].*_.*\.cc"
+echo %relpath% | findstr %find_arg% > nul
+if not errorlevel 1 exit /b
+set "cppfilelist=%cppfilelist% %relpath%"
+exit /b
+
+:find_node
+if exist %config%\node set nodeexe=%config%\node&exit /b
+where node > nul 2> nul
+if not errorlevel 1 set nodeexe=node&exit /b
```

Comment:
This picks up the Node in the path if there isn't a local one right? I guess that's alright for linting, as long as we're not doing it for testing as well.